core/src/main/java/org/apache/druid/timeline/partition/ShardSpec.java

```diff
@@ -134,4 +140,29 @@ default boolean sharePartitionSpace(PartialShardSpec partialShardSpec)
   {
     return !partialShardSpec.useNonRootGenerationPartitionSpace();
   }
+
+  /**
+   * ShardSpec type names.
+   */
+  interface Type
```

Comment:
nit: Access modifier

Reply:
Since the enclosing type is an interface, everything inside is `public` by default.